docs/en/deploy/Running-Alluxio-On-Kubernetes.md

```diff
@@ -320,6 +320,132 @@ will be lost.
+  command: ["/entrypoint.sh"]
+  # --no-format removed from the args
+  args: ["master-only"]
+  ...
+```
+
+**Step 3: Restart Alluxio master and worker Pods**
+
+Now that Alluxio masters and worker containers all use your desired version. Restart them to let it take effect.
+
+Kill the running Alluxio master and worker Pods first.
+
+```console
+$ kubectl delete -f alluxio-worker.yaml
+$ kubectl delete service alluxio-master-0
+$ kubeclt delete statefulset alluxio-master-0
+```
+
+If you have more alluxio-master Pods, delete them together with the corresponding Services.
+
+```console
+$ kubectl delete service alluxio-master-1
+$ kubeclt delete statefulset alluxio-master-1
+$ kubectl delete service alluxio-master-2
+$ kubeclt delete statefulset alluxio-master-2
+```
+
+The reason why you don't want to do `kubectl delete -f alluxio-master.yaml` is that will delete the Persistent Volume Claim in the `alluxio-master.yaml` if any.
+And that will result in your Persistent Volume getting released and the journal in it will be lost.
+
+Now restart the Alluxio master and worker Pods from the YAML files.
+
+```console
+$ kubectl create -f alluxio-master.yaml
+$ kubectl create -f alluxio-worker.yaml
+```
+
+**Step 4: Verify the Alluxio master and worker Pods are back up**
+
+What a proper validation should be varies from case to case.
+The example is just verifying that Alluxio is running.
+
+```console
+# You should see Alluxio master and worker Pods in Running status
+$ kubectl get pods
+# This runs a comprehensive sanity check
+$ kubectl exec <alluxio-master-pod> -- /opt/alluxio/bin/alluxio runTests
+```
+
+#### Rolling Update Alluxio
+
+In a production environment, it is often more desirable to do rolling updates to maintain the availability.
+
+How the rolling update should be done depends on the journal type of Alluxio masters are configured and what storage Alluxio workers are using.
+We will discuss the scenarios separately.
+
+##### Alluxio Masters
+
+The behavior of Alluxio masters will differ depends on which journal mode they are using. 
+See the detailed discussion in [UFS Journal vs Embedded Journal]({{ '/en/operation/Journal.html#ufs-journal-vs-embedded-journal' | relativize_url }})
+
+###### Case A: Rolling Update with Embedded Journal
+
+We do not recommend doing rolling update with embedded journal. 
+
+The reason is Alluxio masters using embedded journal have to reach a consensus before each journaling to its own journal destination.
+The consensus logic of Alluxio masters can be changed in minor versions so if you have a group of Alluxio masters running different versions,
+the behavior will be undefined. 
+
+###### Case B: Rolling Update with UFS Journal
+
+With UFS journal, there is only one shared journal destination and only the one primary master writes. All the secondary masters only read from it.
+
+You are able to do rolling update among Alluxio masters as below.
+
+**Step 1: Update the secondary master Pods and restart them**
```

Comment:
add instruction to identify the primary vs/ secondary (alluxio fs leader command) and kubectl commands to restart